src/js/racingCar.js

```diff
@@ -0,0 +1,144 @@
+      racingCar.MIN_CARNAME_SIZE <= carName.length &&
+      carName.length <= racingCar.MAX_CARNAME_SIZE
+    );
+  }
+
+  participateCarTemplate(raceCarName) {
+    return `
+    <div class="mr-2" data-racecar-name="${raceCarName}">
+      <div class="car-player">${raceCarName}</div>
+    </div>`;
+  }
+
+  participateCarTemplateGenerator() {
+    return this.raceParticipateCar
+      .map((raceCar) => this.participateCarTemplate(raceCar))
+      .join("");
+  }
+
+  openInputNumberOfMatches(e) {
+    e.target.insertAdjacentHTML(
+      "beforeend",
+      `<fieldset>
+        <p>시도할 횟수를 입력해주세요.</p>
+        <div class="d-flex">
+          <input type="number" id="try-count-input" class="w-100 mr-2" placeholder="시도 횟수" />
+          <button id="try-count-btn" class="btn btn-cyan">확인</button>
+        </div>
+      </fieldset>`
+    );
+  }
+
+  matchLoading() {
+    this.raceParticipateCar.forEach((element) => {
+      document
+        .querySelector(`div[data-racecar-name="${element}"]`)
+        .insertAdjacentHTML(
+          "beforeend",
+          `<div class="d-flex justify-center mt-3" data-race-loading="${element}">
+            <div class="relative spinner-container">
+              <span class="material spinner"></span>
+            </div>
+          </div>`
+        );
+    });
+  }
+
+  matchResult() {
+    this.raceParticipateCar.forEach((element) => {
+      document.querySelector(`div[data-race-loading="${element}"]`).remove();
+      if (!this.decidePlay()) return;
+      document
+        .querySelector(`div[data-racecar-name="${element}"]`)
+        .insertAdjacentHTML(
+          "beforeend",
+          `<div class="forward-icon mt-2">⬇️️</div>`
+        );
+    });
+  }
+
+  matchFormGenerator() {
+    document.querySelector("#app").insertAdjacentHTML(
+      "beforeend",
+      `<section class="d-flex justify-center mt-5">
+        <div class="mt-4 d-flex">
+          ${this.participateCarTemplateGenerator()}
+        </div>
+      </section>`
+    );
+  }
+
+  matchProgress() {
+    let count = 1;
+    this.matchLoading();
+    const timeoutId = setInterval(() => {
+      this.matchResult();
+      if (count++ === this.raceCount) {
+        clearInterval(timeoutId);
+        return;
+      }
+      this.matchLoading();
+    }, 1000);
+  }
+
+  submitCarNames(e) {
+    if (!e.target[1].value) {
+      return;
+    }
+    this.raceParticipateCar = e.target[1].value.split(",");
+    this.raceParticipateCar.forEach((element) => {
+      if (!this.testCarNameSize(element)) {
+        alert("5자 이하의 자동차 이름을 입력하세요.");
+        return;
+      }
+    });
+    this.openInputNumberOfMatches(e);
+  }
+
+  submitNumberOfRaces(e) {
+    this.raceCount = e.target[4].valueAsNumber;
+    this.matchFormGenerator();
+    this.matchProgress();
+  }
+
+  gamePrepation = (e) => {
+    e.preventDefault();
+    if (e.submitter.id == "car-name-btn") {
+      this.submitCarNames(e);
+    } else {
+      this.submitNumberOfRaces(e);
+    }
+  };
```

Comment:
이 부분도 질문주신 내용 중에 하나인 것 같은데,
간단하게 해결 방안을 알려드리자면
```suggestion
  step = 0;
  gamePrepation = (e) => {
    e.preventDefault();

    if (this.step === 0) {
      this.submitCarNames(e);
      this.step = 1;
      return;
    }

    this.submitNumberOfRaces(e);
  };
```

이런식으로 표현할 수 있겠네요 ㅎㅎ

이건 굉장히 간략하게 해결 방법을 알려드린거라서, 이 부분을 토대로 리팩토링 해보시면 좋을 것 같아요!
그리고 `시도할 횟수` 입력 후 확인 버튼을 여러번 클릭하는 경우에 대한 처리도 고민도 같이 해보시면 어떨까요?